confdgnmi/src/confd_gnmi_demo_adapter.py

```diff
@@ -207,34 +203,36 @@ def process_changes(self):
                     log.debug("event=%s", event)
                     if event == self.ChangeEvent.ADD:
                         # generate modifications and add them
-                        self.change_db_lock.acquire()
-                        self.adapter.db_lock.acquire()
-                        if "changes" in GnmiDemoServerAdapter.config:
-                            changes = self._get_config_changes()
-                        else:
-                            changes = self._get_random_changes()
-                        send = False
-                        for c in changes:
-                            log.debug("processing change c=%s", c)
-                            if isinstance(c, str):
-                                assert c == "send" or c == "break"
-                                if c == "send":
-                                    send = True
-                                break
-                            else:
-                                log.info("c=%s self.monitored_paths=%s", c, self.monitored_paths)
-                                if any(c[0].startswith(elem) for elem in
-                                       self.monitored_paths):
-                                    log.info("appending c=%s", c)
-                                    self.change_db.append(c)
-                                    if c[0] in self.adapter.demo_db:
-                                        self.adapter.demo_db[c[0]] = c[1]
-                                    elif c[0] in self.adapter.demo_state_db:
-                                        self.adapter.demo_state_db[c[0]] = c[1]
+                        with self.change_db_lock:
+                            with self.adapter.db_lock:
```

Comment:
This is equivalent to
```
with self.change_db_lock, self.adapter.dp_lock:
    ...
```
and you save one indentation level.

Reply:
Will be part of next release